File: app/src/main/java/com/example/miniproject/Database/Converter/SourceModelConverter.java
package com.example.miniproject.Database.Converter;

import androidx.room.TypeConverter;

import com.example.miniproject.Model.SourceModel;

public class SourceModelConverter {
    @TypeConverter
    public static SourceModel toSource(String  id) {
        return id == null ? null : new SourceModel(id);
    }

    @TypeConverter
    public static String toSourceId(SourceModel sourceModel) {
        return sourceModel == null ? null : sourceModel.getId();
    }
}

File: app/src/main/java/com/example/miniproject/database/converter/SourceModelConverter.kt
package com.example.miniproject.database.converter

import androidx.room.TypeConverter
import com.example.miniproject.model.SourceModel

object SourceModelConverter {
    @JvmStatic
    @TypeConverter
    fun toSource(id: String?): SourceModel? {
        return id?.let { SourceModel(it) }
    }

    @JvmStatic
    @TypeConverter
    fun toSourceId(sourceModel: SourceModel?): String? {
        return sourceModel?.id
    }
}